src/main/kotlin/app/revanced/utils/adb/Commands.kt

```diff
@@ -17,7 +17,19 @@ internal fun JadbDevice.buildCommand(command: String, su: Boolean = true): Shell
 }
 
 internal fun JadbDevice.run(command: String, su: Boolean = true): Int {
-    return this.buildCommand(command, su).start().waitFor()
+    if (su) {
+        return this.buildCommand(command, su).start().waitFor()
+    }
+
+    // Avoid deadlock whilst checking for root access
+    val adbCommand = this.buildCommand(command, su).start()
+    val byteArray = ByteArray(512)
```

Comment:
It looks like you're only ever reading a single byte, so instead of allocating 512 bytes, just allocate 1.
```suggestion
    val byteArray = ByteArray(1)
```

Reply:
ByteArray has to be 8 because I read the SU type from it.